### PR TITLE
[Bug](function) make VcompoundPred optimization work well

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes

make VcompoundPred optimization work well
https://github.com/apache/doris/pull/19818 this pr try to enable VcompoundPred optimization but get wrong result on tpcds q28.
The reason is some nullable logic on mysql need special handling.
```sql
mysql [regression_test_tpcds_sf1_p1]>select null and false;
+----------------+
| NULL AND FALSE |
+----------------+
|              0 |
+----------------+
1 row in set (0.00 sec)

mysql [regression_test_tpcds_sf1_p1]>select null and true;
+---------------+
| NULL AND TRUE |
+---------------+
| NULL          |
+---------------+
1 row in set (0.00 sec)

mysql [regression_test_tpcds_sf1_p1]>select null or false;
+---------------+
| NULL OR FALSE |
+---------------+
| NULL          |
+---------------+
1 row in set (0.00 sec)

mysql [regression_test_tpcds_sf1_p1]>select null or true;
+--------------+
| NULL OR TRUE |
+--------------+
|            1 |
+--------------+
1 row in set (0.00 sec)
```

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by yiguolei — COMMENTED** [review]
please add these test to regression test:
mysql [regression_test_tpcds_sf1_p1]>select null and false;
+----------------+
| NULL AND FALSE |
+----------------+
|              0 |
+----------------+
1 row in set (0.00 sec)

mysql [regression_test_tpcds_sf1_p1]>select null and true;
+---------------+
| NULL AND TRUE |
+---------------+
| NULL          |
+---------------+
1 row in set (0.00 sec)

mysql [regression_test_tpcds_sf1_p1]>select null or false;
+---------------+
| NULL OR FALSE |
+---------------+
| NULL          |
+---------------+
1 row in set (0.00 sec)

mysql [regression_test_tpcds_sf1_p1]>select null or true;
+--------------+
| NULL OR TRUE |
+--------------+
|            1 |
+--------------+
1 row in set (0.00 sec)

**Comment by BiteTheDDDDt** [comment]
run buildall

File: regression-test/suites/query_p0/sql_functions/test_predicate.groovy [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                k4 int null
+            )
+            duplicate key (k1)
+            distributed BY hash(k1) buckets 3
+            properties("replication_num" = "1");
+        """
+    sql """ insert into t1 values(1,2,3,null); """
+
+    qt_null_or_null "select * from t1 where abs(k1)=null or abs(k1)=null;"
+    qt_true_or_null "select * from t1 where abs(k1)=1 or abs(k1)=null;"
```
Comment (yiguolei) [inline regression-test/suites/query_p0/sql_functions/test_predicate.groovy:34]: the condition in where predicate is always not null, it is not same as the decription you provide in the issue.

Comment (BiteTheDDDDt) [reply regression-test/suites/query_p0/sql_functions/test_predicate.groovy:34]: > the condition in where predicate is always not null, it is not same as the decription you provide in the issue.

predicate's nullable_mode is depend on arguments

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**